tests/test_thread.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+import concurrent.futures
+
+import pytest
+
+import env  # noqa: F401
+from pybind11_tests import thread as m
+
+
+def test_implicit_conversion():
```

Comment:
Concurrent Futures are not in Python2 so you will need add skips here to make sure they aren't run.

Reply:
Changed to use threading.Thread